src/lib/comms/include/sol-socket.h

```diff
@@ -0,0 +1,352 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <sol-common-buildopts.h>
+#include <sol-network.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup Socket Socket
+ * @ingroup Comms
+ *
+ * @brief API to create an endpoint communication using sockets.
+ *
+ * @{
+ */
+
+struct sol_socket;
+
+/**
+ * @brief The defaults' socket types.
+ */
+enum sol_socket_default_type {
+    SOL_SOCKET_UDP,
+#ifdef SOL_DTLS_ENABLED
+    SOL_SOCKET_DTLS,
+#endif
+};
+
+/**
+ * @brief Available socket options
+ */
+enum sol_socket_option {
+    SOL_SOCKET_OPTION_REUSEADDR,
+    SOL_SOCKET_OPTION_REUSEPORT
+};
+
+/**
+ * @brief The socket option level
+ */
+enum sol_socket_level {
+    SOL_SOCKET_LEVEL_SOCKET,
```

Comment:
The level argument specifies the protocol level at which the option resides (getsockopt man). This is to map get/setsockopt functions.